lib/cli/src/commands/wasmer_create_exe_main.c

```diff
@@ -0,0 +1,172 @@
+  char* error_str = (char*) malloc(error_len);
+  wasmer_last_error_message(error_str, error_len);
+  printf("%s\n", error_str);
+}
+
+#ifdef WASI
+int find_colon(char* string) {
+  int colon_location = 0;
+  for (int j = 0; j < strlen(string); ++j) {
+    if (string[j] == ':') {
+      colon_location = j;
+      break;
+    }
+  }
+  return colon_location;
+}
+
+void pass_mapdir_arg(wasi_config_t* wasi_config, char* mapdir) {
+  int colon_location = find_colon(mapdir);
+  if (colon_location == 0) {
+    // error malformed argument
+    fprintf(stderr, "Expected mapdir argument of the form alias:directory\n");
+    exit(-1);
+  }
+  int dir_len = strlen(mapdir) - colon_location;
+  char* alias = (char*)malloc(colon_location + 1);
+  char* dir = (char*)malloc(dir_len + 1);
+  int j = 0;
+  for (j = 0; j < colon_location; ++j) {
+    alias[j] = mapdir[j];
+  }
+  alias[j] = 0;
+  for (j = 0; j < dir_len; ++j) {
+    dir[j] = mapdir[j + colon_location + 1];
+  }
+  dir[j] = 0;
+
+  wasi_config_mapdir(wasi_config, alias, dir);
+  free(alias);
+  free(dir);
+}
+
+// We try to parse out `--dir` and `--mapdir` ahead of time and process those
+// specially. All other arguments are passed to the guest program.
+void handle_arguments(wasi_config_t* wasi_config, int argc, char* argv[]) {
+  for (int i = 1; i < argc; ++i) {
+    // We probably want special args like `--dir` and `--mapdir` to not be passed directly
+    if (strcmp(argv[i], "--dir") == 0) {
+      // next arg is a preopen directory
+      if ((i + 1) < argc ) {
+        i++;
+        wasi_config_preopen_dir(wasi_config, argv[i]);
+      } else {
+        fprintf(stderr, "--dir expects a following argument specifying which directory to preopen\n");
+        exit(-1);
+      }
+    }
+    else if (strcmp(argv[i], "--mapdir") == 0) {
+      // next arg is a mapdir
+      if ((i + 1) < argc ) {
+        i++;
+        pass_mapdir_arg(wasi_config, argv[i]);
+      } else {
+        fprintf(stderr, "--mapdir expects a following argument specifying which directory to preopen in the form alias:directory\n");
+        exit(-1);
+      }
+    }
+    else if (strncmp(argv[i], "--dir=", strlen("--dir=")) == 0 ) {
+      // this arg is a preopen dir
+      char* dir = argv[i] + strlen("--dir=");
+      wasi_config_preopen_dir(wasi_config, dir);
+    }
+    else if (strncmp(argv[i], "--mapdir=", strlen("--mapdir=")) == 0 ) {
+      // this arg is a mapdir
+      char* mapdir = argv[i] + strlen("--mapdir=");
+      pass_mapdir_arg(wasi_config, mapdir);
+    }
+    else {
+      // guest argument
+      wasi_config_arg(wasi_config, argv[i]);
+    }
+  }
+}
+#endif
+
+int main(int argc, char* argv[]) {
+  wasm_config_t* config = wasm_config_new();
+  wasm_config_set_engine(config, OBJECT_FILE);
+  wasm_engine_t* engine = wasm_engine_new_with_config(config);
+  wasm_store_t* store = wasm_store_new(engine);
+  
+  wasm_module_t* module = wasmer_object_file_engine_new(store, "qjs.wasm");
```

Comment:
I think this is a typo. Can we use the first argument in `argv` (or "wasmer-exe" in case argc == 0)

Reply:
argc should never be 0 I think, there's always supposed to be 1 arg which is the name of the program... We could handle it but I'm not aware of a situation in which that can happen